Feature/1427348 refine UI 2

C2H6Ethan opened the pull request:
[//]: # 'This is the default template for Merge Requests'

## Feature description

**As** an end-user  
**I want to** use a clean UI 
**So that** my eyeballs are satisfied

[//]: # 'Describe how the feature was implemented.'
[//]: # 'Describe new views, workflow actions or logic in relation to the User Story.'
[//]: # 'If the change is technical only, specify how you improved our solution.'

### Changes

| File         | Changes                     |
| ------------ | --------------------------- |
| `CHANGELOG.md` |Added Changes made in this feature to the changelog|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/Footer/Footer.tsx`|Removed the CBQL toggle button from the Footer|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/Footer/footer.cy.tsx`|Removed the CBQL toggle button tests from the Footer tests|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/Query.tsx`|Set the CBQL toggle button to be located to the right of the Query content|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/cbqlToggle/CbqlToggle.tsx`|Moved the CBQL toggle button to its own file|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/cbqlToggle/cbqlToggle.cy.tsx`|Moved the CBQL toggle button tests to its own file|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/queryResult/QueryResult.tsx`|Made the whole QueryResult clickable|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/queryResult/queryResult.css`|Added hover effects which changes the background to a light gray and changes the cursor to a pointer. Also aligned the query result to the center vertically|
|`editorextensions/codebeamer-custom-ui/modal/src/pages/import/import.css`|Added padding to the very top of the the modal and removed a styling due to query changes|
|`editorextensions/codebeamer-custom-ui/src/importmodal.ts`|Changed the generation of coordinates to spread out more depending on how many cards are imported, does not get bigger than 80% of the viewport. This makes sure that small imports don't end up on other sides of the board but also doesn't place a bigger import into one small area.|

## How do you feel about this MR?

good, went kind of overkill on the coordinate generation but at least its kinda neat.
![coordinateGeneration](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark/assets/83122000/a07c63fb-a276-48a8-b62c-e04ebcf36a62)


## Checklist

### Assignee

-   [x] Adequate Milestone set :triangular_flag_on_post:  
-   [x] Adequate labels set :label:  
-   [x] Affected documentation updated :books:
    - [x] Changelog :bookmark_tabs:
    - [ ] Wiki :blue_book:

### Reviewer

-   [ ] Testing
    -   [ ] Coverage :ok_hand:
    -   [ ] Quality :ok_hand:
    -   [ ] Evergreen :100:
-   [ ] Code Quality
    -   [ ] :zero: new lint errors :poop:  
    -   [ ] Maintainable :tools:
    -   [ ] Good patterns :third_place:

C2H6Ethan committed:
Merge pull request #6 from codeBeamer-Extensions-and-Addons/release/v1.0.0

Release/v1.0.0

C2H6Ethan committed:
Merge branch 'main' of https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark into feature/1427348-refine-ui-2

C2H6Ethan committed:
changed coordinate generation

C2H6Ethan committed:
improved coordinate generation by placing the middle of the card on the generated point

C2H6Ethan committed:
updated changelog for the previous change

C2H6Ethan committed:
added padding to the top of the import modal

C2H6Ethan committed:
made coordinate generation more mathimatical

C2H6Ethan committed:
Added hover and made entire result clickable

C2H6Ethan committed:
Changelog: Coordinate and query result updates

C2H6Ethan committed:
created and added cbqlToggle button to Query.tsx

C2H6Ethan committed:
removed cbql toggle button from footer

C2H6Ethan committed:
updated changelog for CBQL toggle button

urecha commented on editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/cbqlToggle/CbqlToggle.tsx:
```diff
@@ -0,0 +1,74 @@
+				onClick={toggleSearchMethod}
+				data-test="search-method"
+			>
+				{advancedSearch ? (
```
I'd argue the expression here should be inverted. You want  ![image](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark/assets/70019423/f1828778-9ff7-4727-a08c-b10b496c78c0) to show when you're on the assisted query, to _switch to CBQL_, and vice-versa. You don't need "confirmation" you're right now on the CBQL input, but to be informed of the alternative.

urecha commented on editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/Query.tsx:
```diff
@@ -13,21 +14,26 @@ export default function Query() {
-	} else {
-		return <CbqlInput />;
-	}
+			)}
+			<CbqlToggle />
+		</div>
```
A few highlights on the CBQLToggle button:
![cb_lucid_ui_stains](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark/assets/70019423/a9366461-436c-42e8-9c29-b0b7fbdcc28a)

I imagine you're aware, but the button changing width with state and its unaligned vertical positioning are a notable stain that we should get rid of, if we're _refining_ the UI in the first place.

Make sure the button dimensions stay consistent, and reconsider the button's placement.  

A single `rem` of `margin-top` happens to be rather nicely centered with the input:
![image](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark/assets/70019423/d4925e17-a678-4b45-9285-b9ec9e8e6cc5)

And to distinguish this button from the rest of the input form, I would think something like a border and some padding like this would be adequate:
![image](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-lucidspark/assets/70019423/fd674b7d-544c-43fa-9e15-d81a70573efd)  
That way, the button doesn't look completely out of place / random.

urecha commented on editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/query/cbqlToggle/CbqlToggle.tsx:
```diff
@@ -0,0 +1,74 @@
+	);
+}
+
+function DefaultOverlayTrigger(props: {
+	children: JSX.Element;
+	content: string;
+}): JSX.Element {
+	return (
+		<OverlayTrigger
+			placement="bottom"
+			trigger={['hover', 'focus']}
+			delay={{ show: 250, hide: 250 }}
+			overlay={
+				<Tooltip className="tooltip-grey">{props.content}</Tooltip>
+			}
+		>
+			{props.children}
+		</OverlayTrigger>
+	);
```
This is now a duplicate in `Footer.tsx` and `CbqlToggle.tsx`, right? Then let's make it a shared component instead.  
Might want to make `placement` a(n optional) `prop`, and then it should be pretty generically usable.

urecha reviewed (changes requested):
Going overkill on coordinate generation is a classic, considering [what we ha](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-miro/blob/master/src/api/utils/getSnailCoords.ts)[d in miro](https://github.com/codeBeamer-Extensions-and-Addons/codebeamer-miro/blob/master/src/api/utils/getConcentricCircleCoords.ts)

C2H6Ethan committed:
inverted advancedSearch expression in CbqlToggle

C2H6Ethan committed:
made the cbqlToggle button the same size

C2H6Ethan committed:
added 1rem margin to the top of cbqltoggle button

C2H6Ethan committed:
added a border before cbqltoggle

C2H6Ethan committed:
moved DefaultOverlayTrigger to its own component

C2H6Ethan committed:
added tests for DefualtOverlayTrigger

C2H6Ethan committed:
updated cbqlToggle.cy.tsx

urecha committed:
fix: CbqlToggle icon not clickable

parameters-icon wasn't classed clickable, making the cursor
default when over it

urecha committed:
feat: inc version nr in extension.ts

To distinguish between dev version and regular one
and also just as a mark of developing the next one
increase the version in the console output to 1.1-alpha
